test/helpers/bundle.js

```diff
@@ -1,28 +1,32 @@
-const isPlainObj = require('is-plain-obj')
-const sinon = require('sinon')
+import { promises as fs } from 'fs'
 
-const { resolveFixtureName } = require('./fixtures')
-const { normalizeHandler, isValidHandler } = require('./handler')
+import isPlainObj from 'is-plain-obj'
+import sinon from 'sinon'
+
+import { resolveFixtureName } from './fixtures.js'
+import { normalizeHandler, isValidHandler } from './handler.js'
 
 // Retrieve Edge Handlers bundled handlers
-const loadBundle = function (t, fixtureName) {
-  const { manifest, bundlePath } = loadManifest(t, fixtureName)
-  const handlers = requireBundle(t, bundlePath)
+export const loadBundle = async function (t, fixtureName) {
+  const { manifest, jsBundlePath } = await loadManifest(t, fixtureName)
+  const handlers = await requireBundle(jsBundlePath)
+  t.true(handlers.every(isValidHandler))
   return { manifest, handlers }
 }
 
 // Load Edge Handlers `manifest.json`
-const loadManifest = function (t, fixtureName) {
+const loadManifest = async function (t, fixtureName) {
   const fixtureDir = resolveFixtureName(fixtureName)
   const localOutDir = `${fixtureDir}/.netlify/edge-handlers`
-  const manifestPath = `${localOutDir}/manifest.json`
-  // eslint-disable-next-line import/no-dynamic-require, node/global-require
-  const manifest = require(manifestPath)
+  const manifestContents = await fs.readFile(`${localOutDir}/manifest.json`, 'utf8')
+  const manifest = JSON.parse(manifestContents)
 
   validateManifest(t, manifest)
 
   const bundlePath = `${localOutDir}/${manifest.sha}`
-  return { localOutDir, manifest, bundlePath }
+  const jsBundlePath = `${bundlePath}.js`
+  await fs.rename(bundlePath, jsBundlePath)
```

Comment:
The Edge handlers plugin produces JavaScript files without any file extension.
However, `import` statements (as opposed to CommonJS `require()`) fail on files without any file extension.
Therefore, we add one during testing.